src/robot_upstart/providers.py

```diff
@@ -34,6 +34,13 @@
 
 from catkin.find_in_workspaces import find_in_workspaces
 
+def detect_provider():
+    cmd=open('/proc/1/cmdline', 'rb').read().split('\x00')[0]
```

Comment:
I should of ran the test before I committed, thanks for catching it. Applied fixed in #47.  Maybe we should enable Travis for this repo?